New service and method API (SHOOP-2293)

suutari-ai opened the pull request:
Refactor payment and shipping methods from a BaseMethodModule based API
to a new polymorphic model based API with behavior components.

Most of these changes are pretty much unsplittable and therefore in a
single commit.  Sorry!

There is also some additions of utilities and some tests which are
separated from the main commit.

Payment and shipping method management from Shop Admin is currently
removed.  New admin modules will follow in other PRs.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1705/
Test PASSed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1706/
Test PASSed.

suutari-ai commented:
Just noticed that Change Log entries are missing. Please don't merge this just yet. Will add the change log entries soon...

tulimaki commented on shoop/core/models/_service_shipping.py:
```diff
@@ -0,0 +1,87 @@
+        verbose_name_plural = _("custom carriers")
+
+    def get_service_choices(self):
+        return [ServiceChoice('manual', _("Manually processed shiment"))]
```
shipment

suutari-ai replied:
Oops.. at least it's not shitment :) Will fix it. Thanks!

tulimaki commented on shoop/core/migrations/0020_services_and_methods.py:
```diff
@@ -0,0 +1,409 @@
+    Link existing ShippingMethods and PaymentMethods which use the
+    previous default module to the created carrier or payment processor.
+
+    Update polymorphic_ctypes for CustomerCarrier and
```
This method does not update polymorphic_ctypes anymore. Can you move the this comment.

tulimaki commented:
This looks good. Mergeable after those comments above is solved. I think we should wait a little bit for the admin-side PRs.

suutari-ai commented:
* Fixed the "shiment" typo.
* Added change log entries.
* Fixed docstrings in migrations.
* Removed admin form provides from shoop.testing, since they don't do anything yet.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1707/
Test PASSed.

anders-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 
http://spartacus.aintra:8080//job/shoop-ng-pull-requests/1708/
Test PASSed.

tulimaki commented:
Great, let's do this!
